### DISCUSSION TITLE
Error: Returning from a request handler without responding or attaching an abort handler is forbidden!

### DISCUSSION THREAD

**Discussion by andymans** [uNetworking/uWebSockets.js#45] [Q&A] [answered]
This server is excellent under many circumstances. 
But when, as part of the request handling, we need to wait for 'something else' to happen, I see some odd behaviour.

e.g. request arrives and we need to fetch some data before responding:

```
get('/my-stuff', (res, req) => {
   db.query('select * from some_table', (err, reply) => {
        res.end(reply)
})
```

or request arrives and we need to call out to an external service in order to satisfy it:

```
get('/auth/twitter', (res, req) => {
   twitterOauth.getRequestToken( (err, token, tokenSecret, results) => {
        if (err) {
            res.end('oops - we could not fetch a token for you')
        } else {
             res.end('append this token to your url')
       }
    })
})
```

At this point the server reports: "Error: Returning from a request handler without responding or attaching an abort handler is forbidden!"

That's pretty interesting. Any ideas how to fix? (nb this may well be 'me' being supremely dopey!)

**Comment by ghost** [comment, marked as answer]
> Error: Returning from a request handler without responding or attaching an abort handler is forbidden!

You are returning from a request handler without responding. So did you attach an abort handler like it tells you to?

examples/VideoStreamer.js

**Reply by kvalood** [reply]
`examples/AsyncFunction.js` That's right?

**Comment by andymans** [comment]
Good info. Thx.

**Comment by s0kil** [comment]
It's a good idea to document this behavior.
